File: exercises/day14-Santa-package-factory-exercises/csharp/PackagingService/PackagingService.cs
namespace PackagingService
{
    public class PackagingService
    {
        public PackageType DeterminePackageType(Gift gift, Child child)
        {
            if (gift.Size == GiftSize.EXTRA_LARGE)
            {
                return PackageType.SPECIAL_CONTAINER;
            }
            
            if (gift.IsFragile)
            {
                return gift.Size == GiftSize.SMALL 
                    ? PackageType.BOX_SMALL 
                    : PackageType.BOX_MEDIUM;
            }
            
            if (child.Age < 5)
            {
                return PackageType.GIFT_BAG;
            }
            
            return gift.Size switch
            {
                GiftSize.SMALL => PackageType.BOX_SMALL,
                GiftSize.MEDIUM => PackageType.BOX_MEDIUM,
                GiftSize.LARGE => PackageType.BOX_LARGE,
                _ => PackageType.WRAPPER_PAPER
            };
        }
        
        public bool CanPackageGift(Gift gift, Child child)
        {
            if (!child.HasBeenNice)
            {
                return false;
            }
            
            if (gift.RecommendedMinAge > child.Age)
            {
                return false;
            }
            
            return true;
        }
    }
}

File: exercises/day14-Santa-package-factory-exercises/kotlin/src/main/kotlin/packagingservice/PackagingService.kt
package packagingservice

class PackagingService {

    fun determinePackageType(gift: Gift, child: Child): PackageType {
        if (gift.size == GiftSize.EXTRA_LARGE) {
            return PackageType.SPECIAL_CONTAINER
        }

        if (gift.isFragile) {
            return if (gift.size == GiftSize.SMALL) {
                PackageType.BOX_SMALL
            } else {
                PackageType.BOX_MEDIUM
            }
        }

        if (child.age < 5) {
            return PackageType.GIFT_BAG
        }

        return when (gift.size) {
            GiftSize.SMALL -> PackageType.BOX_SMALL
            GiftSize.MEDIUM -> PackageType.BOX_MEDIUM
            GiftSize.LARGE -> PackageType.BOX_LARGE
            GiftSize.EXTRA_LARGE -> PackageType.WRAPPER_PAPER
        }
    }

    fun canPackageGift(gift: Gift, child: Child): Boolean {
        if (!child.hasBeenNice) {
            return false
        }

        if (gift.recommendedMinAge > child.age) {
            return false
        }

        return true
    }
}
